packages/lexicon/lightning.d.ts

```diff
@@ -0,0 +1,161 @@
+import type { SanitizePolicies } from '@openfn/logger';
+
+type StepId = string;
+
+/**
+ * Type definitions for Lightning and Worker interfaces
+ *
+ * This is the lightning-worker contract
+ *
+ * It is helpful to have these in the lexicon to avoid a circular dependency between lightning and the worker
+ * It's also kinda nice that the contract isn't in the worker itself, it's on neutral ground
+ */
+// An run object returned by Lightning
+export type Run = {
+  id: string;
+  name?: string;
+  dataclip_id: string;
+  starting_node_id: string;
+
+  triggers: Node[];
+  jobs: Node[];
+  edges: Edge[];
+
+  options?: RunOptions;
+};
+
+/**
+ * These are options that can be sent to the worker with an execution plan
+ * They broadly map to the Workflow Options that are fed straight into the runtime
+ * and saved to the plan itself
+ * (although at the time of writing timeout is handled by the worker, not the runtime)
+ */
+export type RunOptions = {
+  runTimeoutMs?: number;
+  sanitize?: SanitizePolicies;
+  start?: StepId;
+};
+
+// TODO rename to step
+// maybe also split into jobs and triggers
+export type Node = {
+  id: string;
+  name?: string;
+  body?: string;
+  adaptor?: string;
+  credential?: any;
+  credential_id?: string;
+  type?: 'webhook' | 'cron'; // trigger only
+  state?: State;
+};
+
+export interface Edge {
+  id: string;
+  source_job_id?: string;
+  source_trigger_id?: string;
+  target_job_id: string;
+  name?: string;
+  condition?: string;
+  error_path?: boolean;
+  errors?: any;
+  enabled?: boolean;
+}
+
+export type DataClip = Record<string, any>;
+
+export type Credential = Record<string, any>;
+
+export type ExitReasonStrings =
+  | 'success'
+  | 'fail'
+  | 'crash'
+  | 'kill'
+  | 'cancel'
+  | 'exception';
+
+export type CONNECT = 'socket:connect';
+
+// client left or joined a channel
+export type CHANNEL_JOIN = 'socket:channel-join';
+export type CHANNEL_LEAVE = 'socket:channel-leave';
+
+// Queue Channel
+
+// This is the event name
+export type CLAIM = 'claim';
+
+// This is the payload in the message sent to lightning
+export type ClaimPayload = { demand?: number };
+
+// This is the response from lightning
+export type ClaimReply = { runs: Array<ClaimRun> };
+export type ClaimRun = { id: string; token: string };
+
+// Run channel
+
+export type GET_PLAN = 'fetch:plan';
+export type GET_CREDENTIAL = 'fetch:credential';
+export type GET_DATACLIP = 'fetch:dataclip';
+export type RUN_START = 'run:start';
+export type RUN_COMPLETE = 'run:complete';
+export type RUN_LOG = 'run:log';
+export type STEP_START = 'step:start';
+export type STEP_COMPLETE = 'step:complete';
+
+export type ExitReason = {
+  reason: ExitReasonStrings;
+  error_message: string | null;
+  error_type: string | null;
+};
+
+export type GetPlanPayload = void; // no payload
+export type GetPlanReply = Run;
```

Comment:
right, i remember being a little confused by this. when we "getPlan" i'd expect a plan to be returned... but a "Run" is returned. It's not technically a `Run` in lightning's world. To lightning, it's a `RunPlusSomeOtherThings` - is that not exactly what we want to call a `Plan`?

Reply:
Well - yes, you're right! I'll rename it to `LightningPlan` though to help distinguish Lighting's `RunPlusSomeOtherThings` structure from the runtime's `ExecutionPlan` structure.

But in the future Lightning will be sending the `ExecutionPlan` directly (I hope!)